articles/advanced-r-concepts/index.md

```diff
@@ -0,0 +1,333 @@
+
+### Accessing inbuilt Datasets
+At this point, you are in a position to use R effectively. While learning various machine learning algorithms, the best place to start is the in-built datasets. These are small in size and are cleaned thoroughly.
+
+These datasets were the benchmark datasets around a decade ago. Today, budding data scientists use these as tools to learn the various machine learning algorithms. A few of the datasets that R provides are as follows:
+
+1. **Airquality** dataset: Dataset monitoring the air quality and the effect of air quality on weather and temperature. This dataset helps one understand the correlation between air quality and temperature. This [article](https://towardsdatascience.com/a-guide-to-data-visualisation-in-r-for-beginners-ef6d41a34174) goes through exploring this dataset in detail. Go through it for more information.
+```r
+ head(airquality)
+```
+
+ The dataset can be called through the variable `airquality`.
+
+```txt
+ Ozone Solar.R Wind Temp Month Day
+ 1    41     190  7.4   67     5   1
+ 2    36     118  8.0   72     5   2
+ 3    12     149 12.6   74     5   3
+ 4    18     313 11.5   62     5   4
+ 5    NA      NA 14.3   56     5   5
+ 6    28      NA 14.9   66     5   6
+```
+2. **Iris Dataset**: The [iris dataset](https://en.wikipedia.org/wiki/Iris_flower_data_set) provides four features for three species of flowers: *Versonica, Setosa, and Virginica*. The features are the sepal length, sepal width, petal length, and the petal width.
+
+ ```r
+ head(iris)
+ ```
+ Outputs:
+
+ ```txt
+ Sepal.Length Sepal.Width Petal.Length Petal.Width Species
+ 1          5.1         3.5          1.4         0.2  setosa
+ 2          4.9         3.0          1.4         0.2  setosa
+ 3          4.7         3.2          1.3         0.2  setosa
+ 4          4.6         3.1          1.5         0.2  setosa
+ 5          5.0         3.6          1.4         0.2  setosa
+ 6          5.4         3.9          1.7         0.4  setosa
+ ```
+
+
+For a thorough list, refer to the link [mentioned](https://www.rdocumentation.org/packages/datasets/versions/3.6.2).
+
+Pat yourselves on the back for having made this far. From now on, we will work with flow control and modularizing the code written.
+
+### Conditional and Loop Statements
+1. **If-else statements**: If-else statements allows one to implement sequential logic. These are an important aspect of OOP paradigm. Look at the example given below:
+
+    ```r
+    x <- runif(1,0,10)
+    if(x>3){
+        y<-10
+    }
+    else{
+        y <-0
+    }
+    ```
+    The `runif` function randomly picks `n` number of samples in the specified range `[min, max).` `[` denotes closed interval, whereas `)` denotes open interval. The arguments to the function `runif` are:
```

Comment:
What does this mean more concretely? What do the 1, 0, and 10 mean respectively?